modules/sfm/src/fundamental.cpp

```diff
@@ -0,0 +1,580 @@
+                              Mat_<T> _F )
+  {
+    libmv::Mat x1, x2;
+    libmv::Mat3 F;
+
+    cv2eigen(_x1, x1);
+    cv2eigen(_x2, x2);
+
+    libmv::NormalizedEightPointSolver(x1, x2, &F);
+
+    eigen2cv(F, _F);
+  }
+
+  void
+  normalizedEightPointSolver( InputArray _x1, InputArray _x2, OutputArray _F )
+  {
+    const Mat x1 = _x1.getMat(), x2 = _x2.getMat();
+    const int depth = x1.depth();
+    CV_Assert(x1.dims == 2 && x1.dims == x2.dims && (depth == CV_32F || depth == CV_64F));
+
+    _F.create(3, 3, depth);
+
+    Mat F = _F.getMat();
+
+    // type
+    if( depth == CV_32F )
+    {
+      normalizedEightPointSolver<float>(x1, x2, F);
+    }
+    else
+    {
+      normalizedEightPointSolver<double>(x1, x2, F);
+    }
+
+  }
+
+  template<typename T>
+  void
+  relativeCameraMotion( const Mat_<T> &R1,
+                        const Mat_<T> &t1,
+                        const Mat_<T> &R2,
+                        const Mat_<T> &t2,
+                        Mat_<T> R,
+                        Mat_<T> t )
+  {
+    R = R2 * R1.t();
+    t = t2 - R * t1;
+  }
+
+  void
+  relativeCameraMotion( InputArray _R1, InputArray _t1, InputArray _R2,
+                        InputArray _t2, OutputArray _R, OutputArray _t )
+  {
+    const Mat R1 = _R1.getMat(), t1 = _t1.getMat(), R2 = _R2.getMat(), t2 = _t2.getMat();
+    const int depth = R1.depth();
+    CV_Assert((R1.cols == 3 && R1.rows == 3) && (R1.size() == R2.size()));
+    CV_Assert((t1.cols == 1 && t1.rows == 3) && (t1.size() == t2.size()));
+    CV_Assert((depth == CV_32F || depth == CV_64F) && depth == R2.depth() && depth == t1.depth() && depth == t2.depth());
+
+    _R.create(3, 3, depth);
+    _t.create(3, 1, depth);
+
+    Mat R = _R.getMat(), t = _t.getMat();
+
+    // type
+    if( depth == CV_32F )
+    {
+      relativeCameraMotion<float>(R1, t1, R2, t2, R, t);
+    }
+    else
+    {
+      relativeCameraMotion<double>(R1, t1, R2, t2, R, t);
+    }
+
+  }
+
+  template<typename T>
+  void
+  motionFromEssential( const Mat_<T> &_E,
+                       std::vector<Mat> &_Rs,
+                       std::vector<Mat> &_ts )
+  {
+    libmv::Mat3 E;
+    std::vector < libmv::Mat3 > Rs;
+    std::vector < libmv::Vec3 > ts;
+
+    cv2eigen(_E, E);
+
+    libmv::MotionFromEssential(E, &Rs, &ts);
+
+    _Rs.clear();
+    _ts.clear();
+
+    int n = Rs.size();
+    CV_Assert(ts.size() == n);
+
+    for ( int i = 0; i < n; ++i )
+    {
+      Mat_<T> R_temp, t_temp;
+
+      eigen2cv(Rs[i], R_temp);
+      _Rs.push_back(R_temp);
+
+      eigen2cv(ts[i], t_temp);
+      _ts.push_back(t_temp);
+    }
+
+  }
+
+  void
+  motionFromEssential( InputArray _E, OutputArrayOfArrays _Rs,
+                       OutputArrayOfArrays _ts )
+  {
+    const Mat E = _E.getMat();
+    const int depth = E.depth();
+    CV_Assert(E.cols == 3 && E.rows == 3 && (depth == CV_32F || depth == CV_64F));
+
+    std::vector<Mat>& Rs = *(std::vector<Mat>*)_Rs.getObj();
```

Comment:
`getObj()` should not be used here. Use `getMatVector()` instead.
But currently it doesn't support `std::vector<Matx>`.

For more info see `cv::split()` (especially part with set of "create" calls):
https://github.com/Itseez/opencv/blob/14b006e808f4b74e3cba39a4f4b196dc1228bc65/modules/core/src/convert.cpp#L927-L934